extensions/BMO/Extension.pm

```diff
@@ -2514,37 +2512,9 @@ sub _split_crash_signature {
       extract_multiple($crash_signature, [sub { extract_bracketed($_[0], '[]') }])];
 }
 
-sub _fetch_product_version_file {
-  my ($product, $cache_only) = @_;
-  my $key      = "${product}_versions";
-  my $versions = Bugzilla->request_cache->{$key}
-    || Bugzilla->memcached->get_data({key => $key});
-
-  return $versions if $cache_only;
-
-  unless ($versions) {
-    my $ua = Mojo::UserAgent->new;
-    if (my $proxy_url = Bugzilla->params->{'proxy_url'}) {
-      $ua->proxy->http($proxy_url);
-    }
-
-    my $response = $ua->get(PD_ENDPOINT . $key . '.json')->result;
-    $versions = Bugzilla->request_cache->{$key}
-      = $response->is_success ? decode_json($response->body) : {};
-    Bugzilla->memcached->set_data({
-      key   => $key,
-      value => $versions,
-      # Cache for 30 minutes if the data is available, otherwise retry in 5 min
-      expires_in => $response->is_success ? 1800 : 300,
-    });
-  }
-
-  return $versions;
-}
-
 sub _get_product_version {
   my ($product, $channel, $detail) = @_;
-  my $versions = _fetch_product_version_file($product);
+  my $versions = fetch_product_version_file($product);
   my $version = $versions->{PRODUCT_CHANNELS->{$product}->{$channel}->{json_key}};
   return $version if $detail;
```

Comment:
The `return 0 unless $versions;` line should happen before this return.
Also, because `$versions` is a hashref, it'll always be true in this context, that line needs to be `return 0 unless %$versions;`